### ISSUE TITLE
Add ability to filter messages through command in commit_preprocessors

### ISSUE THREAD

**Issue by alerque** [orhun/git-cliff#86]
Issue #84 was about a specific use case and the initial suggestion was a bit of a false start. By the end of the issue discussion I think there was a useful proposal in [this comment](https://github.com/orhun/git-cliff/issues/84#issuecomment-1112106369).

The current use of `commit_precocessors` just allows a regex replacement to run on matching patterns. I propose a new feature that filters the message through an external command. The commit message would be passed on STDIN and a replacement read back from STDOUT. This would allow all sorts of filtering, format conversion, etc.

I think the remaining questions are things like:

* Whether the external command filter should happen before/after/or exclusive of regex replacements. (To my use cases this is a bit of a wash.)
* How exit codes should be considered. (I suggest any errors be considered fatal, a different approach might be to skip commits that error, but I suggest a different mechanism such as skipping anything that returns an empty string but a success exit code.)
* Whether any useful ENV vars should be set that would enable a filter script to do fancier things like read the commit themselves (e.g. maybe setting the current commit sha in an env var would be useful).

**Comment by orhun** [comment]
Hey!

I implemented this in 7d0786ca55423950f0779de4e6a907fc25ae203a:

```toml
[git]
commit_preprocessors = [
	{ pattern = "foo", replace_command = "pandoc -t commonmark"},
	{ pattern = 'bar', replace_command = 'rev | xargs echo "reversed: $@"' },
    { pattern = 'baz', replace_command = 'git show -s --format=%B $COMMIT_SHA' }
]
```

[README.md](https://github.com/orhun/git-cliff#commit_preprocessors) is also updated with these examples.

Feel free to try it out 🐻

**Comment by orhun** [comment]
`0.8.1` is released with this feature.
